src/index/etc/mview.xml

```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0"?>
+<!--
+  ~ Copyright (c) Aligent Consulting. All rights reserved.
+  -->
+
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:Mview/etc/mview.xsd">
+    <view id="fredhopper" class="Aligent\Fredhopper\Model\Indexer\ProductIndexer" group="indexer">
+        <subscriptions>
+            <table name="catalog_product_entity" entity_column="entity_id" />
+            <table name="catalog_product_entity_datetime" entity_column="entity_id" />
+            <table name="catalog_product_entity_decimal" entity_column="entity_id" />
+            <table name="catalog_product_entity_gallery" entity_column="entity_id" />
+            <table name="catalog_product_entity_int" entity_column="entity_id" />
+            <table name="catalog_product_entity_media_gallery_value" entity_column="entity_id" />
+            <table name="catalog_product_entity_text" entity_column="entity_id" />
+            <table name="catalog_product_entity_tier_price" entity_column="entity_id" />
+            <table name="catalog_product_entity_varchar" entity_column="entity_id" />
+            <table name="catalog_product_link" entity_column="product_id"
+                   subscription_model="Aligent\FredhopperIndexer\Model\Mview\View\Attribute\CatalogProductLinkSubscription"/>
+            <table name="catalogrule_product_price" entity_column="product_id" />
+            <table name="cataloginventory_stock_item" entity_column="product_id" />
```

Comment:
Does this cover e.g. `inventory_source_item` as well? Or do we need to add that as well?

Reply:
Subscribing to inventory changes is made more difficult by the lack of a product id in the relevant table(s).
I've had some thoughts, but none is particularly "nice".
The OOTB catalog search implementation is a plugin on the stock indexer that directly calls the catalog search indexer with the changed ids. This is problematic from the point-of-view of interfering with other indexing processes that are potentially happening at the same time.
We could in theory do the same thing, but add to the changelog table instead of directly indexing. This (directly updating the changelog table) just isn't done in OOTB code at all, so it seems a little hacky.